sklearnex/spmd/covariance/tests/test_incremental_covariance_spmd.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from onedal.tests.utils._dataframes_support import (
+    _convert_to_dataframe,
+    get_dataframes_and_queues,
+)
+from sklearnex.tests._utils_spmd import (
+    _generate_statistic_data,
+    _get_local_tensor,
+    _mpi_libs_and_gpu_available,
+)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("assume_centered", [True, False])
+@pytest.mark.mpi
+def test_incremental_covariance_fit_spmd_gold(dataframe, queue, assume_centered):
+    # Import spmd and batch algo
+    from sklearnex.covariance import IncrementalEmpiricalCovariance
+    from sklearnex.spmd.covariance import (
+        IncrementalEmpiricalCovariance as IncrementalEmpiricalCovariance_SPMD,
+    )
+
+    # Create gold data and process into dpt
+    data = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [0.0, 1.0, 2.0],
+            [0.0, 2.0, 4.0],
+            [0.0, 3.0, 8.0],
+            [0.0, 4.0, 16.0],
+            [0.0, 5.0, 32.0],
+            [0.0, 6.0, 64.0],
+        ]
+    )
+
+    local_dpt_data = _convert_to_dataframe(
+        _get_local_tensor(data), sycl_queue=queue, target_df=dataframe
+    )
+
+    # ensure results of batch algo match spmd
+    spmd_result = IncrementalEmpiricalCovariance_SPMD(
+        assume_centered=assume_centered
+    ).fit(local_dpt_data)
+    batch_result = IncrementalEmpiricalCovariance(assume_centered=assume_centered).fit(
+        data
+    )
+
+    assert_allclose(spmd_result.covariance_, batch_result.covariance_)
+    assert_allclose(spmd_result.location_, batch_result.location_)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("num_blocks", [1, 2])
+@pytest.mark.parametrize("assume_centered", [True, False])
+@pytest.mark.mpi
+def test_incremental_covariance_partial_fit_spmd_gold(
```

Comment:
fit should be tested anyway, because it is different interface which can potentially contain errors. I can't get how the second option reducing verbosity. the number of tests would remain the same. moreover, and why do you want to reduce test verbosity?